site/src/nav/DownloadButton.jsx

```diff
@@ -42,49 +42,62 @@ function DownloadButton({ mode, zoom, setZoom }) {
     console.log(bounds);
 
     //Send those to the download engine
-    const minxPath = ["bbox", "minx"];
-    const minyPath = ["bbox", "miny"];
-    const maxxPath = ["bbox", "maxx"];
-    const maxyPath = ["bbox", "maxy"];
+    const xmin = ["bbox", "xmin"];
+    const ymin = ["bbox", "ymin"];
+    const xmax = ["bbox", "xmax"];
+    const ymax = ["bbox", "ymax"];
 
     const readOptions = {
       bbox: bbox,
       bboxPaths: {
-        minxPath,
-        minyPath,
-        maxxPath,
-        maxyPath,
+        xmin,
+        ymin,
+        xmax,
+        ymax,
       },
     };
-
-    let parquetDataset = await new ParquetDataset(DownloadCatalog);
-    const wasmTable = await parquetDataset.read(readOptions);
+    let downloadCatalog = getDownloadCatalog(bbox, visibleTypes);
 
     set_panic_hook();
-    //Create a blob
-    // const binaryDataForDownload = writeGeoParquet(wasmTable);
-    const binaryDataForDownload = writeGeoJSON(wasmTable);
 
-    let blerb = new Blob([binaryDataForDownload], {
-      type: "application/octet-stream",
+    // The catalog contains a base path and then a list of types with filenames. 
+    //First, assemble the parquet datasets in parallel.
+   let datasets = downloadCatalog.types.map(type => {
+      return new ParquetDataset(downloadCatalog.basePath, type.files).then((dataset) => {return {type: type.name, parquet: dataset} });
     });
+      
+    Promise.all(datasets)
+    .then((datasets) => {
+      return datasets.map(dataset => dataset.parquet.read(readOptions).then(reader => {return {type: dataset.type, reader:reader}}));
+    }).then((tableReads) => Promise.all(tableReads)
+    .then((wasmTables) => {
+      wasmTables.map(wasmTable => {
```

Comment:
Sorry to nit, but do you mind fixing the indentation on this block to match the other files? Which reminds me, I don't think we ever merged the prettier PR. Let's do that once our two branches are merged.

Reply:
Yeah, agreed about the indentation here and about the prettier branch. let's get all the features landed for the GA release, then add prettier afterwards, as I expect it will probably change every file in the repo. :)